src/plugins/dashboard/public/dashboard_container/external_api/dashboard_panel_placement_registry.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { PanelPlacementSettings } from '../component/panel_placement/types';
+import { panelPlacementStrings } from '../_dashboard_container_strings';
+
+type GetPanelPlacementSettings<SerializedState extends object = object> = (
+  serializedState?: SerializedState
+) => PanelPlacementSettings;
+
+const registry = new Map<string, GetPanelPlacementSettings<object>>();
+
+export const registerDashboardPanelPlacementSetting = (
+  panelType: string,
```

Comment:
Are panelType and embeddableType the same thing? Should this be embeddableType?

Reply:
They are the same thing. I used `panelType` here to match `panelType` as used in the [`PresentationContainer.addNewPanel` method](https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx#L507). But since we are calling this function from plugins, I changed this to be `embeddableType` which should be less confusing.